modules/widgets/milestone/milestone.php

```diff
@@ -0,0 +1,361 @@
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'event' ); ?>"><?php _e( 'Event' ); ?></label>
+			<input class="widefat" id="<?php echo $this->get_field_id( 'event' ); ?>" name="<?php echo $this->get_field_name( 'event' ); ?>" type="text" value="<?php echo esc_attr( $instance['event'] ); ?>" />
+		</p>
+
+		<fieldset>
+			<legend><?php _e( 'Date and Time' ); ?></legend>
+
+			<label for="<?php echo $this->get_field_id( 'month' ); ?>" class="screen-reader-text"><?php _e( 'Month' ); ?></label>
+			<select id="<?php echo $this->get_field_id( 'month' ); ?>" class="month" name="<?php echo $this->get_field_name( 'month' ); ?>">
+				<?php
+				global $wp_locale;
+				for ( $i = 1; $i <= 12; $i++ ) {
+					$monthnum = zeroise( $i, 2 );
+					echo '<option value="' . esc_attr( $monthnum ) . '"' . selected( $i, $instance['month'], false ) . '>' . $monthnum . '-' . $wp_locale->get_month_abbrev( $wp_locale->get_month( $i ) ) . '</option>';
+				}
+				?>
+			</select>
+
+			<label for="<?php echo $this->get_field_id( 'day' ); ?>" class="screen-reader-text"><?php _e( 'Day' ); ?></label>
+			<input id="<?php echo $this->get_field_id( 'day' ); ?>" class="day" name="<?php echo $this->get_field_name( 'day' ); ?>" type="text" value="<?php echo esc_attr( $instance['day'] ); ?>">,
+
+			<label for="<?php echo $this->get_field_id( 'year' ); ?>" class="screen-reader-text"><?php _e( 'Year' ); ?></label>
+			<input id="<?php echo $this->get_field_id( 'year' ); ?>" class="year" name="<?php echo $this->get_field_name( 'year' ); ?>" type="text" value="<?php echo esc_attr( $instance['year'] ); ?>">
+
+			@ <label for="<?php echo $this->get_field_id( 'hour' ); ?>" class="screen-reader-text"><?php _e( 'Hour' ); ?></label>
+			<input id="<?php echo $this->get_field_id( 'hour' ); ?>" class="hour" name="<?php echo $this->get_field_name( 'hour' ); ?>" type="text" value="<?php echo esc_attr( $instance['hour'] ); ?>">
+
+			<label for="<?php echo $this->get_field_id( 'min' ); ?>" class="screen-reader-text"><?php _e( 'Minutes' ); ?></label>
+			: <input id="<?php echo $this->get_field_id( 'min' ); ?>" class="minutes" name="<?php echo $this->get_field_name( 'min' ); ?>" type="text" value="<?php echo esc_attr( $instance['min'] ); ?>">
+		</fieldset>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'message' ); ?>"><?php _e( 'Message' ); ?></label>
+			<textarea id="<?php echo $this->get_field_id( 'message' ); ?>" name="<?php echo $this->get_field_name( 'message' ); ?>" class="widefat"><?php echo esc_textarea( $instance['message'] ); ?></textarea>
+		</p>
+	</div>
+
+		<?php
+	}
+
+	/**
+	 * Cache the translations, but do it on-demand so it's
+	 * not run every page load whether used or not.
+	 *
+	 * @return array
+	 */
+	public static function get_interval_labels() {
+		static $labels = null;
+
+		// Static variables can't be initialized to arrays on declaration, so we do it here:
+		if ( is_null( $labels ) ) {
+			$labels = array(
+				'year'    => __( 'year' ),
```

Comment:
Probably not necessary to add context to these, Core doesn't -- https://github.com/WordPress/WordPress/blob/4356a28b52530140cc1d57b18c9c23b167a9b491/wp-admin/includes/template.php#L714-L727